### ISSUE TITLE
Detect typing in omnibox

### ISSUE THREAD

**Issue by noblereign** [noblereign/GX-Mods-Extension#4]
Probably will be a little roundabout.
Allow the user to set GXM as a "search engine" (`chrome_settings_overrides, search_provider`) so we can detect typing in the omnibox.
Pass through searches and suggestions to an engine of the user's choice (grab a list using [search api](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/search)).

**Comment by noblereign** [comment]
okay looking at this now this seems like it's gonna be way more complex than i thought 😭
firefox's search api doesn't actually return any of the urls i need (query url and autocomplete url)

i'd probably need to ask the user to input a link to the correct OpenSearch xml description. (for reference, [here's Google's](https://www.google.com/searchdomaincheck?format=opensearch).)

*then*, search urls MUST be an https link. i would rather not host an entire server for this, for longevity and privacy reasons
[LibRedirect](https://github.com/libredirect/browser_extension/) gets around this by giving Firefox a URL that will never lead anywhere, and then *intercepting the network request to redirect it to the correct place.* yikes. that's not ideal but i guess it could work.

but also, would that even be possible for autocomplete? i'd have to try and find out, but worst case scenario we'd need the user to forfeit autocomplete...
also also, guessing i can't change search engine icon at runtime either, so it'd be stuck as either the GXM logo or a generic magnifying glass.

all of these hoops and hurdles to jump through just to get a scuffed search engine passthrough... is it really worth all of this for some typing sounds?
<img src="https://github.com/noblereign/GX-Mods-Extension/assets/2935979/5769f9de-0c02-4872-a3c2-0602c5b231c0" width="256" height="256">

**Comment by noblereign** [comment]
currently trying to figure this out again, this might be even more impossible than i thought it was??

- keyword is absolutely **required** for the extension to detect keystrokes, _regardless of whether or not the extension is currently the default search engine, without a "gx " at the beginning we don't know anything_
- can't detect autocomplete web requests, as those are made on a level higher than extensions can observe. so **we can't detect typing on the default engine OR our own(!!)**
- can't detect keystrokes using `command` api as those require a modifier to be valid (Ctrl or Shift)

**_tl;dr: there are zero ways of detecting even just keypresses in the address bar without forcing people to type a mandatory word at the beginning of every search_**

i genuinely think that webextensions are too limited to implement this in a seamless way. :(
if anyone has any ideas or suggestions feel free to leave them in, i think i have to give up on this for now 💔

**Comment by noblereign** [comment]
if anyone genuinely wants to be typing in "gx `my search term here`" to look things up then i guess i'll implement it?? but i don't really think enough people would be okay with that to warrant me spending time on it

**Comment by noblereign** [comment]
HANG ON. I JUST FOUND SOMETHING. WE MIGHT BE COOKING

**Comment by noblereign** [comment]
"fixed" by #20
